src/zyre_node.c

```diff
@@ -879,19 +879,19 @@ zyre_node_recv_peer (zyre_node_t *self)
             }
         }
 #ifdef ZYRE_BUILD_DRAFT_API
-        // TODO- CURVE- how does this trigger???
-        char public_key[41];
-        if (self->secret_key) {
-            char *value; // i think zhash_destroy will free this one(?)
-            zhash_t *headers = zhash_dup (zre_msg_headers(msg));
-            value = (char *) zhash_lookup (headers, "X-PUBLICKEY");
-            assert (value);
-            strcpy(public_key, value);
-            assert (public_key[0] != 0);
-            zhash_destroy (&headers);
+        if (!self->secret_key) {
```

Comment:
ha, this may have been why you saw the double if(self->secret) .. i may have been working through this and lost track of what i was doing. good catch.